src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.SslProvider.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.Net.Http
+{
+    internal partial class CurlHandler : HttpMessageHandler
+    {
+        private static class SslProvider
+        {
+            private delegate int SslCtxCallback(IntPtr curl, IntPtr sslCtx, IntPtr userPtr);
+
+            private static readonly SslCtxCallback s_sslCtxCallback = SetSslCtxVerifyCallback;
+            private static readonly Interop.Ssl.AppVerifyCallback s_sslVerifyCallback = VerifyCertChain;
+
+            internal static void SetSslOptions(EasyRequest easy)
+            {
+                int answer = Interop.libcurl.curl_easy_setopt(
+                    easy._easyHandle,
+                    Interop.libcurl.CURLoption.CURLOPT_SSL_CTX_FUNCTION,
+                    s_sslCtxCallback);
+
+                switch (answer)
+                {
+                    case Interop.libcurl.CURLcode.CURLE_OK:
+                        break;
+                    case Interop.libcurl.CURLcode.CURLE_NOT_BUILT_IN:
+                        VerboseTrace("CURLOPT_SSL_CTX_FUNCTION is not supported, platform default https chain building in use");
+                        break;
+                    default:
+                        ThrowIfCURLEError(answer);
+                        break;
+                }
+            }
+
+            private static int SetSslCtxVerifyCallback(
+                IntPtr curl,
+                IntPtr sslCtx,
+                IntPtr userPtr)
+            {
+                using (SafeSslContextHandle ctx = new SafeSslContextHandle(sslCtx, ownsHandle: false))
+                {
+                    Interop.Ssl.SslCtxSetCertVerifyCallback(ctx, s_sslVerifyCallback, userPtr);
+                }
+
+                return Interop.libcurl.CURLcode.CURLE_OK;
+            }
+
+            private static int VerifyCertChain(IntPtr storeCtxPtr, IntPtr arg)
```

Comment:
OpenSsl `X509_STORE` and `X509_STORE_CTX` have very little to do with Windows (or .NET) `X509Store`.  They're more like `X509Chain` if looked at one way, or `X509Certificate2Collection` if looked at another so this question doesn't really make sense in the way you mean.

If the problem is a new generally-trusted root that the machine didn't know about, a command such as that may make it work.  But the whole purpose of this change is that the user can add it to their own CurrentUser\Root X509Store and control their trust individually.